### ISSUE TITLE
Can't connect to server

### ISSUE THREAD

**Issue by Luticus** [the-djmaze/snappymail#740]
**SnappyMail version, browser, browser version:**
Snappy mail 2.22.5
Firefox / librewolf 107.0-1 AND chromium 107.0.5304.121
Nextcloud 25.0.1

**Expected behavior and actual behavior:**
Expected is that it logs into my email account
Actual: It does not log into my email account and presents the error "Can't connect to server"
Other email clients work fine. Among them are, Kontact/Kmail, outlook, k9-mail, and Rainloop

Rainloop works currently with the same exact settings as snappymail. Logs in immediately.

**Steps to reproduce the problem:**
configure snappy mail, attempt to login.

**Logs or screenshots:**
![image](https://user-images.githubusercontent.com/7807109/205428037-d17b849d-e1ea-4779-9d0b-57041f9a5654.png)
There doesn't appear to be any other info the the browser log or in the nc log that seem relevant, but I'm looking to see if there are any other detials I can provide. If there's anything that can be attached to help troubleshoot, please let me know and I'll try to get it attached.

**Comment by the-djmaze** [comment]
1. Go Nextcloud -> Admin -> Additional settings.
2. Enable `debug` and save
3. Try login again

Look in the log at `/data/appdata_snappymail/_data_/_default_/logs` to find the issue.

**Comment by Luticus** [comment]
Well for some reason when I check that button in firefox the entire screen goes blank. In chromium it worked, could be a plugin thing or something, haven't tested thoroughly yet.

Anyway, it gets weirder. So when I enable debug it's says authentication error, so I checked my mail server's fail2ban to see if maybe my nextcloud box got caught up somehow, it hasn't. Wouldn't make sense anyway since rainloop works. Whats really weird is that when i go into the admin panel for snappy, and got to domains and then test from there it turns green and works, but when I use the user test on the page just before you click on a domain, it doesn't work. Maybe I'll try removing and adding my domain back in. There's a major disconnect going on here.

**Comment by Luticus** [comment]
The only things I'm getting out of the logs are "bad credentials" but I know that's not true, and deleted and added the domain just to be sure. Also I'm seeing this, which might be relevant:

SERVICE[WARNING]: -  @ /var/www/nextcloud/apps/snappymail/app/snappymail/v/2.22.6/app/libraries/MailSo/Net/NetClient.php#136
[2022-12-05 18:11:28.737][a8c8ceef] [NOTICE]: MailSo\Net\Exceptions\SocketAlreadyConnectedException in /var/www/nextcloud/apps/snappymail/app/snappymail/v/2.22.6/app/libraries/MailSo/Net/NetClient.php:136

**Comment by MI-KY** [comment]
Just an idea because the log says that the socket seems already in use: Rainloop and SnappyMail do not seem to function inside a Nextcloud at the same time... See #583 for some details...
Could you try to disable Rainloop for a moment just to see if SnappyMail then works?

**Comment by Luticus** [comment]
I've tried that in the past and only re-enabled rainloop because snappy wasn't working. Let me test it again though.

**Comment by Luticus** [comment]
I disabled rainloop, removed it, and rebooted the server. Still no dice.

**Comment by Luticus** [comment]
[2022-12-05 22:35:51.999][d1bf3d49] IMAP[WARNING]: MailSo\Imap\Exceptions\NegativeResponseException: AUTHENTICATIONFAILED Authentication failed. in /var/www/nextcloud/apps/snappymail/app/snappymail/v/2.22.6/app/libraries/MailSo/Imap/ResponseCollection.php:46
...
Next MailSo\Imap\Exceptions\LoginBadCredentialsException: AUTHENTICATIONFAILED Authentication failed. in /var/www/nextcloud/apps/snappymail/app/snappymail/v/2.22.6/app/libraries/MailSo/Imap/ImapClient.php:255
...
IMAP[ERROR]: MailSo\Net\Exceptions\SocketAlreadyConnectedException in /var/www/nextcloud/apps/snappymail/app/snappymail/v/2.22.6/app/libraries/MailSo/Net/NetClient.php:136
...
[2022-12-05 22:35:52.043][d1bf3d49] SERVICE[WARNING]: ConnectionError[104]
#0 /var/www/nextcloud/apps/snappymail/app/snappymail/v/2.22.6/app/libraries/RainLoop/Actions/UserAuth.php(142): RainLoop\Actions->CheckMailConnection()

**Comment by Luticus** [comment]
I'm also seeing this on my mail server:  
mail dovecot: imap-login: Disconnected (auth failed, 1 attempts in 2 secs): user=<user@example.com>, method=PLAIN, rip=x.x.x.x, lip=x.x.x.x, TLS, session=<qqqqqqqqqqqqq>

I can assure you the credentials are right and the connection details are also correct.

**Comment by Luticus** [comment]
Also noticed strange behavior:  
![image](https://user-images.githubusercontent.com/7807109/205773473-008feab9-63fe-4c19-bd5e-8d9876df8aa5.png)
This is despite the fact that clicking test inside the "add domain dialogue" lit up green like everything is working fine.

**Comment by MI-KY** [comment]
The "undefined matched domain" message - as far as i know - is no error but a a bit confusing way to say that your domain was matched 🙂. Would need to take a look into the source but also on my system SnappyMail is saying "undefined matched domain".
Because your test inside the domain config works (where you manually insert the credentials) I'm asking myself if maybe you use a automatic logon method to login automatically your Nextcloud user to SnappyMail?

**Comment by the-djmaze** [comment]
> The "undefined matched domain" message - as far as i know - is no error but a a bit confusing way to say that your domain was matched slightly_smiling_face

Correct: https://github.com/the-djmaze/snappymail/commit/6b6aa54b9990bdb9028b932c05b683de80d1af06

**Comment by the-djmaze** [comment]
Best is to start over:
1. Nextcloud -> User -> Additional Settings empty login values
2. Remove app from Nextcloud
3. Delete /data/app_snappymail (or rename as backup)
4. Install app again
5. Nextcloud -> Admin -> Additional Settings -> Open SnappyMail Admin
6. Nextcloud -> Admin -> Additional Settings -> Enable debug
7. Domains: Add your domain
8. Domains: test "login@example.com"
9. Nextcloud -> SnappyMail try login

**Comment by Luticus** [comment]
That's a win, I can log in! Thanks!

I think what happened was that I enabled snappy and was messing with it and noticed an issue with pgp or something to that effect and enabled rainloop to test something and it's possible that when i did that it messed something up. Not sure but a clean slate worked! Thanks so much for the troubleshooting! 

Now that I think about it It's also possible that the password used for the admin -> additional settings -> snappy autologin was fat fingered. I think it was simply saying wrong password and then prompting me for a new one, but then even if you type the new one right, it appears to try to use the auto password again anyway. My suggestion is for an interface update to warn that the auto password is in use on the condition of an authentication fail. Definitely didn't think to check that.

Thanks again!

**Comment by lib-dbr** [comment]
Just a note, I started having the same issue with snappymail unable to connect error on login but the domain test was successful.

When initially testing snappymail, I disabled rainloop in nextcloud. When I was happy with snappymail, I uninstalled the rainloop app and this caused snappymail to no longer work.

I was able to correct the issue by 

1. uninstall snappymail
2. rename /data/app_snappymail
3. install snappymail
4. import rainloop data back into snappymail 
5. rename /data/rainloop-storage
